src/image.c

```diff
@@ -1590,7 +1590,14 @@ gdip_rotate_orthogonal_flip_x (GpImage *image, int angle, BOOL flip_x)
 		return OutOfMemory;
 	}
 
-	source = initial_source_offset + (BYTE *) image->active_bitmap->scan0;
+    int isSurfaceSource = 0;
+    if (image->surface != NULL && gdip_bitmap_format_needs_premultiplication(image)) {
+        isSurfaceSource = 1;
+        source = initial_source_offset + (BYTE*)(cairo_image_surface_get_data(image->surface));
+    }
+    else {
+        source = initial_source_offset + (BYTE *)image->active_bitmap->scan0;
+    }
```

Comment:
please use tabs for indentation to follow the code style in this file.

Reply:
done, commit b72439c2c45eb3f0ebbe62b307330a8796014373